trl/trainer/grpo_trainer.py

```diff
@@ -141,13 +168,43 @@ def data_collator(features):  # No data collation is needed in GRPO
         self.max_prompt_length = args.max_prompt_length
         self.max_completion_length = args.max_completion_length  # = |o_i| in the GRPO paper
         self.num_generations = args.num_generations  # = G in the GRPO paper
-        self.generation_config = GenerationConfig(
-            max_new_tokens=self.max_completion_length,
-            do_sample=True,
-            temperature=args.temperature,
-            num_return_sequences=self.num_generations,
-            pad_token_id=processing_class.pad_token_id,
-        )
+        self.use_vllm = args.use_vllm
+
+        if self.use_vllm:
+            if not is_vllm_available():
+                raise ImportError(
+                    "vLLM is not available and `use_vllm` is set to True. Please install vLLM with "
+                    "`pip install vllm` to use it."
+                )
+            self.generation_config = SamplingParams(
+                n=self.num_generations,  # 2 generations per prompt
+                max_tokens=args.max_completion_length,
+                temperature=args.temperature,
+                # top_k=50,
```

Comment:
Dead code, remove?

Reply:
I uncomment: the goal of these param is to be consistent with transformers generation params